src/lighteval/models/base_model.py

```diff
@@ -345,6 +352,97 @@ def greedy_until_with_logits(
             override_bs=override_bs,
         )
 
+    def greedy_until_multi_turn(
+        self, requests: list[GreedyUntilMultiTurnRequest], override_bs: Optional[int] = None
+    ) -> GenerateMultiTurnReturn:
+        for request in requests:
+            request.stop_sequence = as_list(request.stop_sequence) + [self.tokenizer.eos_token]
+            request.tokenized_context = self.tok_encode(request.context)
+
+        results = []
+
+        # if self.accelerator:
```

Comment:
why aren't the requests sent to the correct gpus by accelerate?

Reply:
I removed the data // for multi turn generation in a first time because the functions we made were not the ones we needed. I just readded it this weekend, it works now and is 8 times faster than original implem :)

Reply:
However, I needed to use more custom code (so not the functions defined in the class for it to work)